dao/mongo.go

```diff
@@ -64,3 +65,33 @@ func (m *Mongo) GetPaymentResource(id string) (*models.PaymentResource, error) {
 
 	return &resource, err
 }
+
+// PatchPaymentResource patches a payment resource from the DB
+func (m *Mongo) PatchPaymentResource(id string, paymentUpdate *models.PaymentResourceData) error {
+	paymentSession, err := getMongoSession()
+	if err != nil {
+		return err
+	}
+	defer paymentSession.Close()
+
+	c := paymentSession.DB("payments").C("payments")
+
+	patchUpdate := make(bson.M)
+
+	// Patch only these fields
+	if paymentUpdate.PaymentMethod != "" {
```

Comment:
Should we error if neither patch-able field is populated?

Reply:
It checks for this on line 165 of service/payment.go 

It does the check there rather than here as it can specify the header as a http.StatusBadRequest rather than the header that's generated by an error in this function which is http.StatusInternalServerError

Reply:
👍